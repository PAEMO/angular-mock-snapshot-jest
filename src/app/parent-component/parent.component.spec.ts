import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { ParentComponent } from './parent.component';
import { ChildComponent } from '../child-component/child.component';
import { MockComponent } from 'ng-mocks';

describe('ParentComponentComponent', () => {
  let component: ParentComponent;
  let fixture: ComponentFixture<ParentComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ParentComponent, MockComponent(ChildComponent)]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(ParentComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should check data from component with snapshot testing', () => {
    expect(component.complexedData).toMatchSnapshot();
  });
});
