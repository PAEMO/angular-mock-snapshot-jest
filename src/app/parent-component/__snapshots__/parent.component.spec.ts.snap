// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParentComponentComponent should render 1`] = `
<app-parent-component>
  <p>
    parent-component works!
  </p><app-child-component
    age="27"
    name="PA"
    ng-reflect-age="27"
    ng-reflect-name="PA"
  /><span>
    Child component is above
  </span>
</app-parent-component>
`;
